draft-ietf-ppm-dap.md

```diff
@@ -1461,6 +1461,16 @@ The Helper responds to the Leader with HTTP status code 201 Created and a body
 consisting of the `AggregationJobResp`, with media type
 "application/dap-aggregation-job-resp".
 
+Changing an aggregation job's parameters is illegal, so further requests to
+`PUT /tasks/{tasks}/aggregation_jobs/{aggregation-job-id}` for the same
+`aggregation-job-id` but with a different `AggregationJobInitReq` in the body
+MUST fail with an HTTP client error status code.
```

Comment:
What requirement are you trying to meet here? To me the subtext reads like "we should allow retries as long as retrying doesn't change the parameters". If so, how about making it more explicit?

Something like this:

> Changing an aggregation job's parameters is illegal. If another request to `PUT /tasks/{tasks}/aggregation_jobs/{aggregation-job-id}` for the same `aggregation-job-id` is made, the Helper MAY ignore the request if the message payload is the same as the previous request. Otherwise, if the message payload has changed or the Helper did not check, the Helper MUST abort the request.

Reply:
The requirement is that once an aggregation job has been initialized, you can't change its parameters, i.e. its aggregation parameter or the partial batch selector. We're not discussing retries anymore.

Reply:
Ah Ok, then it that case, I wouldn't mention the payload explicitly: Just abort if the PUT is repeated for a particular aggregation job.

Reply:
That doesn't work, because it would make it impossible to retry an aggregation job initialization request, which would make DAP basically impossible to operate.

Reply:
We discussed this offline. The intention is to require idempotence here without spelling out how it's supposed to be achieved. There are a few options (quoting @tgeoghegan):
> 1. You could store the entire aggregation job init req that led to the creation of the aggregation job, and then compare the next request.
> 2. You could store a hash of the aggregation job init req and check it against hashes of next request.
> 3. You could reevaluate the aggregation job init req and check if its results match what you have stored from the last time you initialized the request.

I'm a little uncomfortable about leaving this up to implementations, especially since (3.) is a bit fiddly, but it's probably less fiddly to try to spell it out.